include/vapor/VaporField.h

```diff
@@ -1,28 +1,29 @@
 #ifndef VAPORFIELD_H
 #define VAPORFIELD_H
 
-#include <list>
 #include "vapor/Field.h"
 #include "vapor/Particle.h"
 #include "vapor/DataMgr.h"
 #include "vapor/FlowParams.h"
 #include "vapor/Grid.h"
-#include "vapor/unique_ptr_cache.hpp"
+#include "vapor/ptr_cache.hpp"
 
 namespace flow {
 
 //
 // Helper class: it is used to identify a specific grid.
 //
-class FLOW_API GridKey final {
+class FLOW_API GridKey {
 private:
+    uint32_t              _timestep = std::numeric_limits<uint32_t>::max(); // almost impossible value
+    int32_t               _refLev = -2;   // Impossible value
+    int32_t               _compLev = -2;  // Impossible value
     std::string           _varName;
-    uint64_t              _timestep;
-    int32_t               _refLev, _compLev;
-    std::array<double, 3> _ext_min, _ext_max;
+    VAPoR::CoordType      _ext_min = {0.0, 0.0, 0.0};
+    VAPoR::CoordType      _ext_max = {0.0, 0.0, 0.0};
 
 public:
-    void Reset(uint64_t, int32_t, int32_t, std::string, const std::vector<double> &, const std::vector<double> &);
+    void Reset(uint32_t, int32_t, int32_t, std::string, VAPoR::CoordType, VAPoR::CoordType);
```

Comment:
Why switch from passing const references to non-const CoordType values?

Reply:
In this case, I was attempting to make the function signature simpler (i.e., without using const and references).

Const references avoid a copy, which can be expensive if 1) the copy happens on the heap, and 2) the copy is big. In this case, after changing the signature, a copy happens on the stack and quite small with just 3 values, so it's not expensive anyways. I think the simplicity wins in this case.

Reply:
Gotcha.  I was suprised to see this but if there's not a big performance hit then readability wins.

Reply:
Yeah, references have its own overheads, so for small amounts of data (debatable how small is small), simply make a copy seems easiest for everyone (including the compiler).